### PR TITLE
Add ajv validation

### PR DESCRIPTION
**Pull request by JamesMilnerUK**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [ ] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Uses ajv instead of `jsonschema`

screenshot_20190128_224651

### CODE REVIEW

File: src/commands/validate.ts [hunk 1]
```diff
@@ -73,7 +211,7 @@ export function builtInCommandValidation(validation: ValidationWrapper): Promise
 		const commandKey = `${commandGroup}-${commandName}`; // group and name are required properties
 
 		if (validation.commandConfig === undefined) {
-			logSchemaErrors(`.dojorc config does not have the top level command property '${commandKey}'`);
+			logSchemaTopLevelError(commandKey);
```
Comment (JamesMilnerUK) [inline src/commands/validate.ts]: I've just realised if there is no config at all (i.e. no `.dojorc` or one in the `package.json`) this branch will trigger. The question is do we want that behaviour? If people have no `.dojorc` we probably should be trying to validate at all.

**Comment by agubler** [comment]
@JamesMilnerUK What is the status of this?

**Comment by JamesMilnerUK** [comment]
@agubler I think it's 90% there. I think as I left a comment above it tries to validate when no config is found which is probably undesirable. There's a pretty comprehensive test suite there, but might benefit from some more smoke testing with some real project configs.

**Comment by agubler** [comment]
@JamesMilnerUK yup a smoke test for commands with validations would help (with a `.dojorc` and also a `package.json` dojo config), also agree that it should not validate when no configuration is found.

**Pull request merged**